treeherder/seta/job_priorities.py

```diff
@@ -0,0 +1,142 @@
+    # all TaskCluster high value jobs.
+    task_request = _update_task_request(project)
+    last_reset = task_request.seconds_since_last_reset()
+    job_priorities = []
+    for jp in _query_job_priorities(priority=0, excluded_build_system_type='buildbot'):
+        # We only add a job if it hasn't reached its timeout.
+        # If the timeout is zero it means that the job needs to always run
+        if last_reset < jp.timeout or jp.timeout == 0:
+            # Due to the priority of all high value jobs is 1, and we
+            # We need to return all jobs for every 5th push.
+            if task_request.counter % jp.priority != 0:
+                job_priorities.append(jp)
+
+    return _process(build_system='taskcluster', job_priorities=job_priorities)
+
+
+def _update_task_request(project):
+    '''Update a repository's counter and/or reset timer.'''
+    now = datetime.datetime.now()
+
+    try:
+        task_request = TaskRequest.objects.get(repository__name=project)
+        # Increase the counter of how many times the gecko decision task
+        # has contacted us
+        task_request.counter += 1
+        if task_request.has_expired():
+            task_request.last_reset = now
+        task_request.save()
+    except TaskRequest.DoesNotExist:
+        task_request = TaskRequest.objects.create(
+            repository=Repository.objects.get(name=project),
+            counter=1,
+            last_reset=now,
+            reset_delta=RESET_DELTA)
+
+    return task_request
+
+
+def _query_job_priorities(priority, excluded_build_system_type):
+    job_priorities = JobPriority.objects.all()
+    if priority != 0:
+        job_priorities = job_priorities.filter(priority=priority)
+    return job_priorities.exclude(buildsystem=excluded_build_system_type)
+
+
+def _validate_request(build_system_type, project):
+    if build_system_type not in ('buildbot', 'taskcluster', '*'):
+        raise SetaError('Valid build_system_type values are buildbot or taskcluster.')
+
+    if project not in ('mozilla-inbound', 'autoland', 'mozilla-central', 'graphics'):
+        raise SetaError('We currently do not support {}.'.format(project))
+
+
+def seta_job_scheduling(project, build_system_type, priority=5, user_agent=None):
+    _validate_request(build_system_type, project)
+
+    if user_agent == GECKO_DECISION_TASK_USER_AGENT:
+        ref_data_names = _gecko_decision_task_request(project)
+    else:
+        excluded_build_system_type = 'taskcluster' if build_system_type == 'buildbot' else 'buildbot'
+        job_priorities = _query_job_priorities(priority, excluded_build_system_type)
+        ref_data_names = _process(build_system_type, job_priorities)
+
+    # XXX: I don't think we really need 'jobtypes' and today's date in the returning data
+    #      Getting rid of it will require the consumers to not expect it.
+    return {'jobtypes': {str(datetime.date.today()): sorted(ref_data_names)}}
```

Comment:
I agree we do not need the date- as we are building this for TC primarily, lets get V1 with the current format, and maybe make a V2 API which removes the date- then we can migrate BB to use v1, and maybe v2 if it works out smoothly.